### PR TITLE
BACKPORT: groupby.apply incorrectly dropping nan #43236

### PR DESCRIPTION
**Pull request by debnathshoham**
xref #43236

### CODE REVIEW

File: doc/source/whatsnew/v1.3.3.rst [hunk 1]
```diff
@@ -17,6 +17,8 @@ Fixed regressions
 - Fixed regression in :class:`DataFrame` constructor failing to broadcast for defined :class:`Index` and len one list of :class:`Timestamp` (:issue:`42810`)
 - Performance regression in :meth:`core.window.ewm.ExponentialMovingWindow.mean` (:issue:`42333`)
 - Fixed regression in :meth:`.GroupBy.agg` incorrectly raising in some cases (:issue:`42390`)
+- Fixed regression in :meth:`.GroupBy.apply` where ``nan`` values were dropped even with ``dropna=False`` (:issue:`43205`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.3.3.rst:20]: umm you added L21 which is not part of this PR

Comment (debnathshoham) [reply doc/source/whatsnew/v1.3.3.rst:20]: my bad.. removed

**Comment by jreback** [comment]
thanks @debnathshoham
